src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -705,9 +705,12 @@ else if (type == TokenTypes.VARIABLE_DEF) {
         else if (type == TokenTypes.IDENT) {
             visitIdentToken(ast, variablesStack);
         }
-        else if (isInsideLocalAnonInnerClass(ast)) {
-            final TypeDeclDesc obtainedClass = getSuperClassOfAnonInnerClass(ast);
-            modifyVariablesStack(obtainedClass, variablesStack, ast);
+        else {
+            final DetailAST lastChild = ast.getLastChild();
+            if (lastChild != null && lastChild.getType() == TokenTypes.OBJBLOCK) {
+                final TypeDeclDesc obtainedClass = getSuperClassOfAnonInnerClass(ast);
+                modifyVariablesStack(obtainedClass, variablesStack, lastChild);
+            }
```

Comment:
From a quick look, this can cause problems as `isInsideLocalAnonInnerClass(#)` returns `false` for type declarations and `lastChild != null && lastChild.getType() == TokenTypes.OBJBLOCK` is `true` for type declarations. 

Tests cases might not be there, but it may cause problems in stack order, etc.

Reply:
I'm curious about a real test case which can catch this. Could you help to come up with at least one such a case? Otherwise I don't think it's an issue.

Reply:
@Vyom-Yadav ping